Modules/System/Permission Sets/src/PermissionSetRelation.Codeunit.al

```diff
@@ -10,6 +10,24 @@ codeunit 9855 "Permission Set Relation"
 {
     Access = Public;
 
+    /// <summary>
+    /// Adds a new permission set relation.
+    /// </summary>
+    /// <param name="AppId">The app ID of the permission set.</param>
+    /// <param name="RoleId">The role ID of the permission set.</param>
+    /// <param name="Scope">The scope of the permission set.</param>
+    /// <param name="RelatedAppId">The app ID of the related permission set.</param>
+    /// <param name="RelatedRoleId">The role ID of the related permission set.</param>
+    /// <param name="RelatedScope">The scope of the related permission set.</param>
+    /// <param name="PermissionType">The type of the new permission set relation.</param>
+    /// <returns>true if creating the permission set relation was successful</returns>
+    procedure AddNewPermissionSetRelation(AppId: Guid; RoleId: Code[30]; Scope: Option System,Tenant; RelatedAppId: Guid; RelatedRoleId: Code[30]; RelatedScope: Option System,Tenant; PermissionType: Option Include,Exclude) Success: Boolean
+    var
+        PermissionSetRelationImpl: Codeunit "Permission Set Relation Impl.";
+    begin
+        exit(PermissionSetRelationImpl.AddNewPermissionSet(AppId, RoleId, Scope, RelatedAppId, RelatedRoleId, RelatedScope, PermissionType));
```

Comment:
Calling this function directly bypasses the security checks of VerifyUserCanEditPermissionSet.

Doesn't calling ModifyPermissionSet achieve the same result you're looking for?

Reply:
Hi Darrick,
Thanks your review with a sharp eye.
Unfortunately ModifyPermissionSet does not achieve the same result I'm looking for.

I created a new procedure AddNewPermissionSetRelation in the "Permission Set Relation Impl.".
According the existing pattern the "Permission Set Relation" AddNewPermissionSetRelation now calls "Permission Set Relation Impl." AddNewPermissionSetRelation.